influxql/statement_rewriter.go

```diff
@@ -56,6 +108,45 @@ func rewriteShowMeasurementsStatement(stmt *ShowMeasurementsStatement) (Statemen
 	}, nil
 }
 
+func rewriteShowMeasurementCardinalityStatement(stmt *ShowMeasurementCardinalityStatement) (Statement, error) {
+	// Check for time in WHERE clause (not supported).
+	if HasTimeExpr(stmt.Condition) {
+		return nil, errors.New("SHOW MEASUREMENT CARDINALITY doesn't support time in WHERE clause")
+	}
+
+	// Use all measurements, if zero.
+	if len(stmt.Sources) == 0 {
+		stmt.Sources = Sources{
+			&Measurement{Regex: &RegexLiteral{Val: regexp.MustCompile(`.+`)}},
+		}
+	}
+
+	return &SelectStatement{
+		Fields: []*Field{
+			{
+				Expr: &Call{
+					Name: "count",
+					Args: []Expr{
+						&Call{
+							Name: "distinct",
+							Args: []Expr{&VarRef{Val: "_name"}},
+						},
+					},
+				},
+				Alias: "count",
+			},
+		},
+		Sources:    stmt.Sources,
+		Condition:  stmt.Condition,
+		Dimensions: stmt.Dimensions,
+		Offset:     stmt.Offset,
+		Limit:      stmt.Limit,
+		OmitTime:   true,
+		Dedupe:     true,
+		IsRawQuery: true,
```

Comment:
This doesn't look like a raw query. Further, doesn't the distinct take care of any deduped points?